File: Power_of_Two.java
public class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n == 0) return false;
        int base = (int)(Math.log(1.0 * n) / Math.log(2.0));
        return n == (int)Math.pow(2.0, base);
    }
}

File: Swift/power-of-two.swift
/**
 * https://leetcode.com/problems/power-of-two/
 * 
 * 
 */ 
class Solution {
    func isPowerOfTwo(_ n: Int) -> Bool {
        var n = n
        while n > 1 {
            if n % 2 != 0 { return false }
            n /= 2
        }
        return n == 1
    }
}
